File: AurigainLoanERP/AurigainLoanERP.Data/ContractModel/AgentModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AurigainLoanERP.Data.ContractModel
{
    public class AgentModel
    {
        public AgentModel() {           
            BankDetails = new UserBankDetailModel1();
            UserKYC = new UserKycModel1();
            UserNominee = new UserNomineeModel1();
        }
        public long Id { get; set; }
        public long UserId { get; set; }
        public string FullName { get; set; }
        public string FatherName { get; set; }
        public string UniqueId { get; set; }
        public string Gender { get; set; }
        public int QualificationId { get; set; }
        public string Address { get; set; }      
        public int? DistrictId { get; set; }
        public string PinCode { get; set; }
        public DateTime? DateOfBirth { get; set; }
        public string ProfilePictureUrl { get; set; }
        public bool? IsActive { get; set; }
        public bool IsDelete { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime? ModifiedOn { get; set; }
        public long? CreatedBy { get; set; }
        public long? ModifiedBy { get; set; }       
        public  UserBankDetailModel1 BankDetails { get; set; }
        public  UserKycModel1 UserKYC { get; set;}
        public  UserNomineeModel1  UserNominee { get; set; }
    }
    public class UserBankDetailModel1
    {
        public long Id { get; set; }
        public string BankName { get; set; }
        public string AccountNumber { get; set; }
        public string Ifsccode { get; set; }
        public string Address { get; set; }
        public long UserId { get; set; }
        public bool? IsActive { get; set; }        
    }
    public class UserNomineeModel1
    {
        public long Id { get; set; }
        public string NamineeName { get; set; }
        public string RelationshipWithNominee { get; set; }
        public long UserId { get; set; }
        public bool? IsActive { get; set; }       
    }
    public class UserKycModel1
    {
        public long Id { get; set; }
        public string Kycnumber { get; set; }
        public int KycdocumentTypeId { get; set; }
        public long UserId { get; set; }
        public bool? IsActive { get; set; }        
    }
}

File: AurigainLoanERPApp/src/app/Shared/Model/Agent/agent.model.ts
import { UserPostModel, UserNomineePostModel, UserReportingPersonPostModel, DocumentPostModel, UserBankDetailsPostModel, UserKYCPostModel, DocumentViewModel, ReportingPersonViewModel, UserBankDetailViewModel, UserKYCViewModel, UserNomineeViewModel, UserSecurityDepositViewModel, UserViewModel } from "../doorstep-agent-model/door-step-agent.model";

export class Agent {
}
export interface AgentListModel {
  Id: number;
  UserId: number;
  RoleId: number;
  Role: string;
  Email: string;
  Mobile: string;
  FullName: string;
  FatherName: string;
  UniqueId: string;
  Gender: string;
  QualificationName: string;
  Address: string;
  DistrictName: string;
  StateName: string;
  PinCode: string;
  DateOfBirth: string;
  ProfilePictureUrl: string;
  IsActive: boolean;
  IsApproved: boolean;
  Mpin: string;
  IsDelete: boolean;
  CreatedOn: string;
  ModifiedOn: string;
  CreatedBy: number;
  ModifiedBy: number;
  ReportingPersonName:string;
  ReportingPersonUserId:number;
}


export class AgentPostModel {
  Id!: number;
  FullName!: string;
  FatherName!: string;
  Gender!: string;
  QualificationId!: number;
  Address!: string;
  AddressLine2!:string;
  StateId!:number;
  DistrictId!: number | null;
  PinCode!: string;
  AreaPincodeId!: number;
  DateOfBirth!: string | null;
  IsActive!: boolean | null;
  User!: UserPostModel;
  Documents: DocumentPostModel[] = [];
  BankDetails!: UserBankDetailsPostModel;
  UserKYC: UserKYCPostModel[] = [];
  UserNominee!: UserNomineePostModel;
  ReportingPerson!: UserReportingPersonPostModel;
}


export interface AgentViewModel {
  Id: number;
  FullName: string;
  FatherName: string;
  UniqueId: string;
  Gender: string;
  QualificationId: number;
  QualificationName: string;
  Address: string;
  DistrictId: number | null;
  DistrictName: string;
  StateId : number ;
  StateName: string;
  PinCode: string;
  AreaPincodeId: number;
  DateOfBirth: string | null;
  ProfilePictureUrl: string;
  IsActive: boolean | null;
  IsDelete: boolean;
  CreatedOn: string;
  ModifiedOn: string | null;
  CreatedBy: number | null;
  ModifiedBy: number | null;
  Documents: DocumentViewModel[] | null;
  User: UserViewModel;
  BankDetails: UserBankDetailViewModel | null;
  UserKYC: UserKYCViewModel[] | null;
  UserNominee: UserNomineeViewModel | null;
  ReportingPerson: ReportingPersonViewModel | null;
}
